011-quarkus-panache-rest-flyway/src/test/java/io/quarkus/qe/PostgreSqlApplicationResourceTest.java

```diff
@@ -76,38 +80,96 @@ public void shouldReturnBadRequestIfApplicationNameIsNull() {
+        thenApplicationsCountIs(0);
+    }
+
+    @Disabled("This filter is ignored. Reported by https://github.com/quarkusio/quarkus/issues/13831")
+    @Test
+    public void shouldFindApplicationWhenFilteringByServiceName() {
+        givenApplicationWithService("service-a");
+        whenFilterApplicationByName("useServiceByName", "name", "service-a");
+        thenApplicationsCountIs(1);
+    }
+
+    @Disabled("This filter is ignored. Reported by https://github.com/quarkusio/quarkus/issues/13831")
+    @Test
+    public void shouldNotFindApplicationWhenFilteringByServiceName() {
+        givenApplicationWithService("service-a");
+        whenFilterApplicationByName("useServiceByName", "name", "service-not-exist");
+        thenApplicationsCountIs(0);
+    }
+
+    private void givenApplicationWithService(String serviceName) {
+        whenCreateApplication("my-app-name");
+        whenCreateService(serviceName);
+    }
+
+    private void whenCreateService(String serviceName) {
+        ServiceEntity service = new ServiceEntity();
+        service.name = serviceName;
+        service.application = actualApplicationEntity;
+        actualServiceEntities.add(servicePath().body(service).post()
+                .then().statusCode(HttpStatus.SC_CREATED).and().extract().as(ServiceEntity.class));
+    }
+
+    private void whenFilterApplicationByName(String filterName, String paramName, String paramValue) {
+        String params = String.format("?%s=%s", paramName, paramValue);
+
+        actualList = applicationPath().body(actualApplicationEntity).get("filterBy/" + filterName + params).then()
+                .statusCode(HttpStatus.SC_OK).and().extract().as(ApplicationEntity[].class);
+    }
+
     private void whenCreateApplication(String appName) {
         ApplicationEntity request = new ApplicationEntity();
         request.name = appName;
         request.version = new VersionEntity();
         request.version.id = EXPECTED_VERSION;
-        actualEntity = applicationPath().body(request).post()
+
+        actualApplicationEntity = applicationPath().body(request).post()
                 .then()
                 .statusCode(HttpStatus.SC_CREATED)
                 .and().extract().as(ApplicationEntity.class);
     }
 
     private void whenUpdateApplication(String appName) {
-        actualEntity.name = appName;
-        applicationPath().body(actualEntity).put("/" + actualEntity.id)
+        actualApplicationEntity.name = appName;
+        applicationPath().body(actualApplicationEntity).put("/" + actualApplicationEntity.id)
                 .then().statusCode(HttpStatus.SC_NO_CONTENT);
     }
 
     private void whenDeleteApplication() {
-        applicationPath().delete("/" + actualEntity.id)
+        for (ServiceEntity service : actualServiceEntities) {
+            servicePath().delete("/" + service.id)
+                    .then().statusCode(HttpStatus.SC_NO_CONTENT);
+        }
```

Comment:
This is a workaround as I would have expected that deleting an application should have deleted also the services... but this is not working (I tried with all options for Cascade).
@rsvoboda do you think this is an issue worthy to be reported?

Reply:
Finally, I've reported an issue: https://github.com/quarkusio/quarkus/issues/13842

Reply:
@Sgitario Please add reference to quarkusio/quarkus#13842 as comment with TODO prefix for the future reference

Reply:
In the end, I didn't need this workaround as I found an approach to filter collections without needing the `@JoinTable` annotation. 
Though I'm monitoring the issue I reported with the limitation of usage of both `@OneToMany` and `@JoinTable` annotations.